services/director-v2/src/simcore_service_director_v2/api/routes/computations_tasks.py

```diff
@@ -0,0 +1,173 @@
+        # Unexpected error: Cannot determine the cause of failure
+        # to get donwload link and cannot handle it automatically.
+        # Will treat it as "not available" and log a warning
+        log_file_url = None
+        log.warning(
+            "Failed to get log-file of %s: %s.",
+            f"{user_id=}/{project_id=}/{node_id=}",
+            err,
+        )
+    return log_file_url
+
+
+# ROUTES HANDLERS --------------------------------------------------------------
+
+
+@router.get(
+    "/{project_id}/tasks/-/logfile",
+    summary="Gets computation task logs file after is done",
+    response_model=list[TaskLogFileGet],
+)
+async def get_all_tasks_log_files(
+    user_id: UserID,
+    project_id: ProjectID,
+    comp_pipelines_repo: CompPipelinesRepository = Depends(
+        get_repository(CompPipelinesRepository)
+    ),
+    comp_tasks_repo: CompTasksRepository = Depends(get_repository(CompTasksRepository)),
+) -> list[TaskLogFileGet]:
+    """Returns download links to log-files of each task in a computation.
+    Each log is only available when the corresponding task is done
+    """
+    # gets computation task ids
+    info = await analyze_pipeline(project_id, comp_pipelines_repo, comp_tasks_repo)
+
+    return [
+        TaskLogFileGet(
+            task_id=node_id,
+            download_link=await _get_task_log_file(user_id, project_id, node_id),
+        )
+        for node_id in (t.node_id for t in info.filtered_tasks)
+    ]
+
+
+@router.get(
+    "/{project_id}/tasks/{node_uuid}/logfile",
+    summary="Gets computation task logs file after is done",
+    response_model=TaskLogFileGet,
+)
+async def get_task_logs_file(
+    user_id: UserID,
+    project_id: ProjectID,
+    node_uuid: NodeID,
+    comp_tasks_repo: CompTasksRepository = Depends(get_repository(CompTasksRepository)),
+) -> TaskLogFileGet:
+    """Returns a link to download logs file of a give task.
+    The log is only available when the task is done
+    """
+
+    if not await comp_tasks_repo.check_task_exists(project_id, node_uuid):
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND,
+            detail=[f"No task_id={node_uuid} found under computation {project_id}"],
+        )
+
+    return TaskLogFileGet(
+        task_id=node_uuid,
+        download_link=await _get_task_log_file(user_id, project_id, node_uuid),
+    )
+
+
+# NOTE: added here as reference for near futuer extensions
+#
+# @router.get(
+#    "/{project_id}/tasks/{node_uuid}/logs",
+#    summary="Gets computation task log",
+# )
+# - Implement close as possible to https://docs.docker.com/engine/api/v1.41/#operation/ContainerTop
+async def get_task_logs(
+    user_id: UserID,
+    project_id: ProjectID,
+    node_uuid: NodeID,
+) -> str:
+    """Gets ``stdout`` and ``stderr`` logs from a computation task. It can return a list of the tail or stream live"""
+    raise NotImplementedError(f"/{project_id=}/tasks/{node_uuid=}/logs")
```

Comment:
if the function is not used, could you comment it out as well?

Reply:
what difference would it make commented or not?
- I think is well documented 
- if used it raises NotImplemented
- I have a test that intentionally fails as a reminder ...

Reply:
The comment on the API part looked a bit strange to me. I would leave it uncommented. I know it is strange to have and available endpoint replying not implemented.